src/Timer.zig

```diff
@@ -0,0 +1,41 @@
+const std = @import("std");
+const builtin = @import("builtin");
+
+const Timer = @This();
+
+backing_timer: BackingTimerType = undefined,
+
+// TODO: verify declarations and its signatures
+const BackingTimerType = if (builtin.cpu.arch == .wasm32) void else std.time.Timer;
+
+/// Initialize the timer.
+pub fn start() !Timer {
+    return Timer{
+        .backing_timer = try BackingTimerType.start(),
+    };
+}
+
+/// Reads the timer value since start or the last reset in nanoseconds.
+pub fn readPrecise(timer: *Timer) u64 {
+    return timer.backing_timer.read();
+}
+
+/// Reads the timer value since start or the last reset in seconds.
+pub fn read(timer: *Timer) f32 {
+    return @intToFloat(f32, timer.readPrecise()) / @intToFloat(f32, std.time.ns_per_s);
+}
+
+/// Resets the timer value to 0/now.
+pub fn reset(timer: *Timer) void {
+    timer.backing_timer.reset();
+}
+
+/// Returns the current value of the timer in nanoseconds, then resets it.
+pub fn lapPrecise(timer: *Timer) u64 {
+    return timer.backing_timer.lap();
+}
+
+/// Returns the current value of the timer in seconds, then resets it.
+pub fn lap(timer: *Timer) f32 {
```

Comment:
BTW, probably we should mark all "wrapper" functions like this with `inline` just to be on the safe side?

Reply:
Alright, I will revisit it soon.

Reply:
Seems like a wise decision; probably shouldn't leave it up to the inline quota.